external-crates/move/crates/move-package/src/compilation/build_plan.rs

```diff
@@ -68,9 +69,16 @@ impl BuildPlan {
             root: resolution_graph.root_package(),
             sorted_deps,
             resolution_graph,
+            compiler_vfs_root: None,
         })
     }
 
+    pub fn set_compiler_vfs_root(mut self, vfs_root: VfsPath) -> Self {
+        assert!(self.compiler_vfs_root.is_none());
+        self.compiler_vfs_root = Some(vfs_root);
+        self
+    }
```

Comment:
I find this very unsatisfying, but am unsure what to do.

Reply:
This could be `std::mem::replace` where you hand back the old one and make the caller deal with it